--- a/js/src/old-configure
+++ b/js/src/old-configure
@@ -2595,10 +2595,10 @@
 fi
 
 
-if test -n "$JS_STANDALONE"; then
-JS_LIBRARY_NAME="mozjs-$MOZILLA_SYMBOLVERSION"
+if test -n "$MOZ_DEBUG"; then
+JS_LIBRARY_NAME="mozjs$MOZILLA_SYMBOLVERSION-debug"
 else
-JS_LIBRARY_NAME="mozjs"
+JS_LIBRARY_NAME="mozjs$MOZILLA_SYMBOLVERSION-release"
 fi
 JS_CONFIG_LIBS="$NSPR_LIBS $LIBS"
 if test "$CC_TYPE" != clang-cl ; then
